Added an appdata.xml file for Linux software gallery integration

Mailaender opened the pull request:
From http://linkdupont.fedorapeople.org/sources/freeorion.appdata.xml which should be included upstream. See https://fedoraproject.org/wiki/Packaging:AppData for more information about what this does.

Mailaender committed:
Create freeorion.appdata.xml

which adds metadata for Linux software galleries.

Mailaender committed:
Install the freeorion.appdata.xml

Mailaender committed:
Update freeorion.appdata.xml

I assume type="default" should only be used once.

Mailaender committed:
Add a missed indent for the description paragraph.

Mailaender committed:
Use the new metainfo directory for appdata.

adrianbroher reviewed (changes requested):
Most of those notes are reminders for myself @Mailaender, no need to act unless you want to.

adrianbroher committed:
Use non-legacy appdata component type

adrianbroher committed:
Use reverse domain notation for appdata id

adrianbroher committed:
Sync appdata description with README

adrianbroher committed:
Convert appdata keywords to categories

adrianbroher committed:
Rename linux desktop integration files

Both *.appdata.xml and *.desktop files should use and reverse domain
identifier according to upstream specifications.

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic
https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id

adrianbroher committed:
Add launchable to appstream file

adrianbroher committed:
Add CC-BY-SA-3.0 to project licence in appstream file

When looking at the game as a single release artifact both licenses
apply.
